src/ucp/wireup/select.c

```diff
@@ -1615,27 +1618,34 @@ ucp_wireup_add_bw_lanes(const ucp_wireup_select_params_t *select_params,
             dev_index         = context->tl_rscs[rsc_index].dev_index;
             sinfo->path_index = dev_count.local[dev_index];
 
-
         } else {
             /* disqualify/count lane_desc_idx */
             addr_index      = select_ctx->lane_descs[excl_lane].addr_index;
             rsc_index       = select_ctx->lane_descs[excl_lane].rsc_index;
             dev_index       = context->tl_rscs[rsc_index].dev_index;
             excl_lane       = UCP_NULL_LANE;
+            skip_dev_index  = dev_index;
         }
 
         /* Count how many times the LOCAL device is used */
         iface_attr = ucp_worker_iface_get_attr(ep->worker, rsc_index);
         ++dev_count.local[dev_index];
-        if (dev_count.local[dev_index] >= iface_attr->dev_num_paths) {
+
+        spare = !!(bw_info->criteria.num_paths_override &&
+                   ((skip_dev_index != UCP_NULL_RESOURCE) &&  /* clang sanitizer */
+                    (skip_dev_index == dev_index)));
+
```

Comment:
Maybe define `num_path` var abd us it like below? And then compare to this value rather than taking the value from `iface_attr`
```suggestion
        if ((bw_info->criteria.num_paths_override &&
            ((skip_dev_index != UCP_NULL_RESOURCE) &&  /* clang sanitizer */
            (skip_dev_index == dev_index))) {
                num_path = iface_attr->dev_num_paths + 1;
            } else {
                num_path = iface_attr->dev_num_paths;
            }

```

Reply:
fixed with local and remote `dev_num_paths`.